src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/ReadYamlStep.java

```diff
@@ -55,13 +59,41 @@
  */
 public class ReadYamlStep extends AbstractFileOrTextStep {
 
+	public static final String DEFAULT_MAX_ALIASES_PROPERTY = ReadYamlStep.class.getName() + ".DEFAULT_MAX_ALIASES_FOR_COLLECTIONS";
+	public static /*almost final*/ int DEFAULT_MAX_ALIASES_FOR_COLLECTIONS = Integer.getInteger(DEFAULT_MAX_ALIASES_PROPERTY, -1);
+	public static final String MAX_MAX_ALIASES_PROPERTY = ReadYamlStep.class.getName() + ".MAX_MAX_ALIASES_FOR_COLLECTIONS";
+	public static /*almost final*/ int MAX_MAX_ALIASES_FOR_COLLECTIONS = Integer.getInteger(MAX_MAX_ALIASES_PROPERTY, 1000);
```

Comment:
Maybe this is waaay to high?

Reply:
If the library default is 50, what is the reason for this to be higher than 50?

Reply:
Because it is the entire reason for this feature to allow someone to make it higher than 50. But you don't want to allow it to be too high or you will get the billion laughs.

Reply:
Ok, there might have been a misunderstanding. Looking at the code, I take it that:

- `-1` is the default if nothing set. This translates to using the default (50) in the library
- a custom `DEFAULT_MAX_ALIASES_FOR_COLLECTIONS` value can be configured
- the value cannot, however, exceed the `MAX_MAX_ALIASES_FOR_COLLECTIONS` value of 100 to avoid potential billion laughs

If that's the case, I'm okay with 100 as the upper guard rail.

Reply:
I guess it's a balancing act between setting the value high enough to catch the edge cases without an administrators approval, but low enough to avoid any real risk of an attack.

Hoping that given 1000 is 1/1000000th of the 3GB produced in the lol example, this would be enough.

Reply:
> Because it is the entire reason for this feature to allow someone to make it higher than 50.

Administrators should need to opt into allowing a higher-than-default value.

Right now, the limit that can be specified by less trusted/untrusted folks able to provide sandboxed Jenkinsfiles is 20x the library default.

Reply:
So, we keep the `MAX_MAX_ALIASES_FOR_COLLECTIONS` at 50? I'm easy either way. The fact that it is possible is the most important feature here.

Reply:
> there might have been a misunderstanding

Maybe I'm still misreading this?

To expand on the behavior I would prefer to see:

1. The library default should also be the default max legal value for the new parameter that can be set by Pipeline authors. I.e., by default, this does nothing.
2. Admins can raise the maximum legal value that can be set with the new parameter by Pipeline authors.
3. Ideally admins should be prevented from shooting themselves in the foot by not allowing _too_ crazy values to not impact Jenkins stability too badly, but that's of secondary importance to me.

AFAIUI, the current implementation does not satisfy item 1 and 3. While there's a limit of what Pipeline authors can set, it's 20x the library default, and there's no limit to what admins can allow. One of the two constants allows admins to change the default limit without parameter, which effectively creates an invisible environment dependency for Jenkinsfiles, making them less portable.